### PR TITLE
flow: Add component for vsphere integration

### LINKED ISSUE #2322: Flow: add vsphere integration component
Add a or more components for the `integrations-next` [vsphere integration](https://grafana.com/docs/agent/next/configuration/integrations/integrations-next/vsphere-config/).

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

This PR adds Flow component for vSphere (vmware) exporter.

#### Which issue(s) this PR fixes

Fixes #2322 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.vsphere.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Omitted fields take their default values.
+
+| Name                         | Type      | Description                                                                                                                             | Default | Required |
+| ---------------------------- | --------- | --------------------------------------------------------------------------------------------------------------------------------------- | ------- | -------- |
+| `vsphere_url`                | `string`  | The url of the vCenter endpoint SDK     |         | no      |
+| `vsphere_user`             | `string` | vCenter username. |    | no       |
+| `vsphere_password`           | `secret` | vCenter password.   |    | no       |
+| `request_chunk_size`         | `int`     | Number of managed objects to include in each request to vsphere when fetching performance counters.                                     | `256`   | no       |
+| `collect_concurrency`        | `int`     | Number of concurrent requests to vsphere when fetching performance counters.                                                           | `8`     | no       |
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.vsphere.md:32]:
```suggestion
| `collect_concurrency`        | `int`     | Number of concurrent requests to vSphere when fetching performance counters.                                                           | `8`     | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.vsphere.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+| `discovery_interval` | `duration` | Interval on which to run vsphere managed object discovery. | `0` | no |
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.vsphere.md:33]:
```suggestion
| `discovery_interval` | `duration` | Interval on which to run vSphere managed object discovery. | `0` | no |
```

File: pkg/integrations/install/install.go [hunk 3]
```diff
@@ -31,6 +31,7 @@ import (
 	_ "github.com/grafana/agent/pkg/integrations/snowflake_exporter"     // register snowflake_exporter
 	_ "github.com/grafana/agent/pkg/integrations/squid_exporter"         // register squid_exporter
 	_ "github.com/grafana/agent/pkg/integrations/statsd_exporter"        // register statsd_exporter
+	_ "github.com/grafana/agent/pkg/integrations/vmware_exporter"        // register vmware_exporter
```
Comment (tpaschalis) [inline pkg/integrations/install/install.go:34]: We also need to change the import path of the v2 integration below on line 47.

But do we know if there's a reason why the integration was v2-only?

Comment (marctc) [reply pkg/integrations/install/install.go:34]: Not really, it was originally added my @rlankfo

**Comment by gloriasc** [comment]
Great work @marctc - thanks for the reviews @tpaschalis and @clayton-cornell 👏

**Pull request merged**